### PR TITLE
T21528

### PR DESCRIPTION
**Pull request by tchx84**
https://phabricator.endlessm.com/T21528

### CODE REVIEW

File: js/app/modules/layout/topMenu.js [hunk 1]
```diff
@@ -8,6 +8,14 @@ const Gtk = imports.gi.Gtk;
 const Module = imports.app.interfaces.module;
 const SlidingPanelOverlay = imports.app.widgets.slidingPanelOverlay;
 
+const DIRECTION = {
```
Comment (ptomato) [inline js/app/modules/layout/topMenu.js]: Coding style nitpick: Use `Direction.DOWN` and `Direction.UP`.

File: js/app/modules/layout/topMenu.js [hunk 2]
```diff
@@ -8,6 +8,14 @@ const Gtk = imports.gi.Gtk;
...
+    UP: 0,
+    DOWN: 1,
+};
+
+const DISTANCE_UP = 20;
```
Comment (ptomato) [inline js/app/modules/layout/topMenu.js]: Clarify these with a comment, or give them names that indicate what the distance represents, or both. Maybe you could make something that's indexable by a value of `Direction`? e.g.
```js
const APPEAR_THRESHOLD_PX = {
    [Direction.UP]: 20,
    [Direction.DOWN]: 200,
};
```

File: js/app/modules/layout/topMenu.js [hunk 3]
```diff
@@ -66,6 +74,10 @@ var TopMenu = new Module.Class({
         this._menu_panel = this.add_panel_widget(top_menu_container, Gtk.PositionType.TOP);
         this._menu_panel.expand = true;
 
+        this._previous_value = 0;
```
Comment (ptomato) [inline js/app/modules/layout/topMenu.js]: I'd name this `_previous_position` to avoid the generic `value`

File: js/app/modules/layout/topMenu.js [hunk 4]
```diff
@@ -82,7 +91,7 @@ var TopMenu = new Module.Class({
         let content = this.create_submodule('content');
         this._grid.add(content);
 
-        if (content instanceof Gtk.ScrolledWindow) {
+        if (this.hide_on_scroll && content instanceof Gtk.ScrolledWindow) {
```
Comment (ptomato) [inline js/app/modules/layout/topMenu.js]: I don't think this check is correct, it's cutting out other logic if `hide_on_scroll` is turned off.

File: js/app/modules/layout/topMenu.js [hunk 5]
```diff
@@ -14,7 +14,7 @@ const DIRECTION = {
 };
 
 const DISTANCE_UP = 20;
-const DISTANCE_DOWN = 200;
+const DISTANCE_DOWN = 275;
```
Comment (ptomato) [inline js/app/modules/layout/topMenu.js]: The code says 275, but the commit message 270. Either seems fine, but they should agree.

Comment (tchx84) [reply js/app/modules/layout/topMenu.js]: Will check with @katelk28 , but for now I will assume the code is correct.

**Pull request merged**